pkg/api/authn.go

```diff
@@ -975,6 +982,104 @@ func GenerateAPIKey(uuidGenerator guuid.Generator, log log.Logger,
 	return apiKey, apiKeyID.String(), err
 }
 
+// extractIdentityFromSource attempts to extract identity from a specific source.
+func extractIdentityFromSource(cert *x509.Certificate, source string, mtlsConfig *config.MTLSConfig) (string, error) {
+	// Normalize to lowercase for case-insensitive matching
+	normalizedSource := strings.ToLower(strings.TrimSpace(source))
+
+	switch normalizedSource {
+	case "commonname", "cn":
+		if cert.Subject.CommonName == "" {
+			return "", fmt.Errorf("%w", zerr.ErrNoIdentityInCommonName)
+		}
+
+		return cert.Subject.CommonName, nil
+
+	case "subject", "dn":
+		return cert.Subject.String(), nil
+
+	case "url", "uri":
+		if len(cert.URIs) == 0 {
+			return "", fmt.Errorf("%w", zerr.ErrNoURISANFound)
```

Comment:
Since there is no other error/info here, would it be possible to directly return the error instead of wrapping it again?

Similar question for Line 1022, 1032, and 1046 below.